Update mandelbrot-asm.js to use boolean vectors.

stoklund opened the pull request:
The SIMD.js spec (v0.9) now has boolean vectors, and
Float32x4.lessThanOrEqual() returns a Bool32x4 vector.

Update the mandelPixelX4() function to expect this new return type from the
comparison.

Commit:
Update mandelbrot-asm.js to use boolean vectors.

The SIMD.js spec (v0.9) now has boolean vectors, and
Float32x4.lessThanOrEqual() returns a Bool32x4 vector.

Update the mandelPixelX4() function to expect this new return type from the
comparison.

bnjbvr commented on js/mandelbrot-asm.js:
```diff
@@ -138,16 +142,16 @@ function asmjsModule (global, imp, buffer) {
       z_re4   = f4add(c_re4, new_re4);
       z_im4   = f4add(c_im4, new_im4);
-      count4  = i4add(count4, i4and(mi4, one4));
+      count4  = i4add(count4, i4sel(mb4, one4, zero4));
```
It's kind of weird to have to do that, but the API doesn't provide any other boolean => integer bit conversion function.

stoklund replied:
Yes, the boolean vector types are a bit more abstract than a 1-1 instruction mapping. We will need to pattern-match things like this to simpler code. For example, `i4sel(mb4, one4, zero4)` should be transformed to `i4and(mb4, one4)` in the JIT's internal representation.

Or, if we want to get clever, map `i4add(count4, i4sel(mb4, one4, zero4))` to `i4sub(count4, mb4)`.

bnjbvr commented:
Thank you for the update! Note that mandelbrot-worker-asm.js also needs an update. Can you add these changes, please?

Commit:
Update mandelbrot-worker-asm.js to use boolean vectors.

Commit:
Update mandelbrot.js and mandelbrot-worker.js.

- The SIMD type names are now capitalized.
- Use the Bool32x4 type returned from lessThanOrEqual.

bnjbvr commented:
Oops, didn't see you updated the patch: merging. Thank you for the update!
